src/main/scala/vectorpipe/osm/internal/PlanetHistory.scala

```diff
@@ -94,13 +94,16 @@ private[vectorpipe] object PlanetHistory {
         /* Y2K bug here, except it won't manifest until the end of the year 1 billion AD */
         val nextTime: Instant = rest.headOption.map(_.meta.timestamp).getOrElse(Instant.MAX)
 
+        /* Nodes associated with this way */
+        val ns = nodes.filter(n => w.nodes.contains(n.meta.id))
```

Comment:
`ns  == nodes` here. Apologies for not making this clearer: the `ways` and `nodes` lists coming into `linesAndPolys` are already associated. The Ways are versions of *one* Way, and the Nodes are all versions of all Nodes ever associated with that Way. 

Given that, is the PR still necessary?

Reply:
They're already associated, but not with individual ways (as I read it); does reducing the set of candidate nodes when reconstructing this particular way help?

Reply:
Forgive my ignorance about Spark, but does the `filter` not provide a benefit by reducing the number of times the full `nodes` list needs to be iterated over in `recentNodes()`?

Reply:
We're not in Spark-land here. Notice the inputs are just `List`.